source/math.d.ts

```diff
@@ -0,0 +1,231 @@
+*/
+// TODO: Support big integer and negative number.
+export type Add<A extends number, B extends number> = [
+	IsEqual<A, PositiveInfinity>, IsEqual<A, NegativeInfinity>,
+	IsEqual<B, PositiveInfinity>, IsEqual<B, NegativeInfinity>,
+] extends infer R extends [boolean, boolean, boolean, boolean]
+	? Or<
+	And<IsEqual<R[0], true>, IsEqual<R[3], false>>,
+	And<IsEqual<R[2], true>, IsEqual<R[1], false>>
+	> extends true
+		? PositiveInfinity
+		: Or<
+		And<IsEqual<R[1], true>, IsEqual<R[2], false>>,
+		And<IsEqual<R[3], true>, IsEqual<R[0], false>>
+		> extends true
+			? NegativeInfinity
+			: true extends R[number]
+				? number
+				: ([IsNegative<A>, IsNegative<B>] extends infer R
+					? [false, false] extends R
+						? [...BuildTuple<A>, ...BuildTuple<B>]['length']
+						: [true, true] extends R
+							? number
+							: ArrayMax<[NumberAbsolute<A>, NumberAbsolute<B>]> extends infer Max_
+								? ArrayMin<[NumberAbsolute<A>, NumberAbsolute<B>]> extends infer Min_ extends number
+									? Max_ extends A | B
+										? Subtract<Max_, Min_>
+										: number
+									: never
+								: never
+					: never) & number
+	: never;
+
+/**
+Returns the difference between two input numbers.
+
+Note:
+- A or B can only support `-999` ~ `999`.
+- If the result is negative, you can only get `number`.
+
+@example
+```
+import type {Subtract} from 'type-fest';
+
+Subtract<333, 222>;
+//=> 111
+
+Subtract<111, -222>;
+//=> 333
+
+Subtract<-111, 222>;
+//=> number
+
+Subtract<PositiveInfinity, 9999>;
+//=> PositiveInfinity
+
+Subtract<PositiveInfinity, PositiveInfinity>;
+//=> number
+```
+
+@category Numeric
+*/
+// TODO: Support big integer and negative number.
+export type Subtract<A extends number, B extends number> = [
+	IsEqual<A, PositiveInfinity>, IsEqual<A, NegativeInfinity>,
+	IsEqual<B, PositiveInfinity>, IsEqual<B, NegativeInfinity>,
+] extends infer R extends [boolean, boolean, boolean, boolean]
+	? Or<
+	And<IsEqual<R[0], true>, IsEqual<R[2], false>>,
+	And<IsEqual<R[3], true>, IsEqual<R[1], false>>
+	> extends true
+		? PositiveInfinity
+		: Or<
+		And<IsEqual<R[1], true>, IsEqual<R[3], false>>,
+		And<IsEqual<R[2], true>, IsEqual<R[0], false>>
+		> extends true
+			? NegativeInfinity
+			: true extends R[number]
+				? number
+				: [IsNegative<A>, IsNegative<B>] extends infer R
+					? [false, false] extends R
+						? BuildTuple<A> extends infer R
+							? R extends [...BuildTuple<B>, ...infer R]
+								? R['length']
+								: number
+							: never
+						: Lt<A, B> extends true
+							? number
+							: [false, true] extends R
+								? Add<A, NumberAbsolute<B>>
+								: Subtract<NumberAbsolute<B>, NumberAbsolute<A>>
+					: never
+	: never;
+
+// TODO: Export this publicly.
+/**
+Returns a boolean for whether A is greater than B.
+
+@example
+```
+Gt<1, -5>;
+//=> true
+
+Gt<1, 1>;
+//=> false
+
+Gt<1, 5>;
+//=> false
+```
+*/
+export type Gt<A extends number, B extends number> = number extends A | B
```

Comment:
Should be named `GreaterThan`. Same with the others.

It would also be nice to publicly export these, but could also be done later. If exported publicly, they should be in separate files.

Reply:
Okay, I'll take care of it later today

Reply:
done